modules/misc/gtk.nix

```diff
@@ -166,14 +166,35 @@ in {
     home.packages = optionalPackage cfg.font ++ optionalPackage cfg.theme
       ++ optionalPackage cfg.iconTheme;
 
-    home.file.${cfg2.configLocation}.text =
-      concatStringsSep "\n" (mapAttrsToList formatGtk2Option ini) + "\n"
-      + cfg2.extraConfig;
+    home.file.${cfg2.configLocation} = {
+      text = concatStringsSep "\n" (mapAttrsToList formatGtk2Option ini) + "\n"
+        + cfg2.extraConfig;
+      onChange = let xsettingsd = "${pkgs.xsettingsd}/bin/xsettingsd";
+      in ''
+        DISPLAY=$(grep "declare -x DISPLAY=" \
+        "''${XDG_RUNTIME_DIR:-/run/user/$UID}/env-vars" \
+        | cut -d '=' -f2 | cut -d '"' -f2) \
+        $DRY_RUN_CMD ${xsettingsd} -c \
+        <( echo "Net/ThemeName \"${cfg.theme.name}\"" ) & \
+        process=$!
+        sleep 0.2 && kill $process || true
+        unset process
+      '';
+    };
 
     home.sessionVariables.GTK2_RC_FILES = cfg2.configLocation;
 
-    xdg.configFile."gtk-3.0/settings.ini".text =
-      toGtk3Ini { Settings = ini // cfg3.extraConfig; };
+    xdg.configFile."gtk-3.0/settings.ini" = {
+      text = toGtk3Ini { Settings = ini // cfg3.extraConfig; };
+      onChange = let
+        gsettings-schemas =
+          "${pkgs.gsettings-desktop-schemas}/share/gsettings-schemas/gsettings-desktop-schemas-${pkgs.gsettings-desktop-schemas.version}";
+      in ''
+        XDG_DATA_DIRS="$XDG_DATA_DIRS":${escapeShellArg gsettings-schemas} \
+        DBUS_SESSION_BUS_ADDRESS="unix:path=''${XDG_RUNTIME_DIR:-/run/user/$UID}/bus" \
+        $DRY_RUN_CMD ${pkgs.glib}/bin/gsettings set org.gnome.desktop.interface gtk-theme ${cfg.theme.name}
```

Comment:
According to the unit tests in place, the theme can be null.

>  error: value is null while a set was expected, at /home/runner/work/home-manager/home-manager/modules/misc/gtk.nix:195:93
(use '--show-trace' to show detailed location information)